contracts/contracts/auth/SiweAuth.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+
+import {SignatureRSV, A13e} from "./A13e.sol";
+import {ParsedSiweMessage, SiweParser} from "../SiweParser.sol";
+import {Sapphire} from "../Sapphire.sol";
+
+struct Bearer {
+    string domain; // [ scheme "://" ] domain.
+    address userAddr;
+    uint256 validUntil; // in Unix timestamp.
+}
+
+/**
+ * @title Base contract for SIWE-based authentication
+ * @notice Inherit this contract, if you wish to enable SIWE-based
+ * authentication for your contract methods that require authenticated calls.
+ * The smart contract needs to be bound to a domain (passed in constructor).
+ *
+ * #### Example
+ *
+ * ```solidity
+ * contract MyContract is SiweAuth {
+ *   address private _owner;
+ *
+ *   modifier onlyOwner(bytes calldata bearer) {
+ *     if (authMsgSender(bearer) != _owner) {
+ *       revert("not allowed");
+ *     }
+ *     _;
+ *   }
+ *
+ *   constructor(string memory domain) SiweAuth(domain) {
+ *     _owner = msg.sender;
```

Comment:
I wonder if it would be clearer if `onlyOwner` were also extended or duplicated to `onlyManager` as well?

Reply:
That would just add clutter IMO. The example code is the minimal code to demonstrate the usage of SiweAuth.